### PR TITLE
Get correct id from shortUrl regardless of the scheme

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?

Properly get the id from a short url regardless of the scheme
## What is the value of this and can you measure success?

CAPI can start returning short urls with https
## Does this affect other platforms - Amp, Apps, etc?

No
## Request for comment

@mchv @johnduffell @sndrs

### CODE REVIEW

File: common/app/model/meta.scala [hunk 1]
```diff
@@ -159,7 +159,14 @@ final case class Fields(
   sensitive: Option[Boolean],
   legallySensitive: Option[Boolean]
 ){
-  def javascriptConfig: Map[String, JsValue] = Map(("shortUrl", JsString(shortUrl)))
+  lazy val shortUrlId = shortUrl.replaceFirst("^[a-zA-Z]+://gu.com", "") //removing scheme://gu.com
```
Comment (TBonnin) [inline common/app/model/meta.scala]: Maybe CAPI should return the short Id as a field 😉  😉  😉

File: identity/app/services/PlaySavedArticlesService.scala [hunk 2]
```diff
@@ -35,9 +35,9 @@ class PlaySavedArticlesService @Inject()(api: IdApiClient) extends SafeLogging w
 
     val sanitizedArticles = savedArticles.articles map {
       article =>
-        val id = article.id.replace("http://www.theguardian.com/","")
-        val shortUrl = article.shortUrl.replace("http://gu.com","")
-        SavedArticle(id, shortUrl, article.date, article.read, article.platform)
+        val id = article.id.replaceFirst("^[a-zA-Z]+://www.theguardian.com/","")
+        val shortUrlId = article.shortUrl.replaceFirst("^[a-zA-Z]+://gu.com","")
```
Comment (TBonnin) [inline identity/app/services/PlaySavedArticlesService.scala]: I wish I could centralize the replace in one place but it looks like SavedArticle have their own model

File: common/app/model/meta.scala (hunk 1 above)
Comment (mchv) [inline common/app/model/meta.scala]: @TBonnin agreed. I am planning to do this

**Comment by mchv** [comment]
👍

**Comment by johnduffell** [comment]
great good work! 👍

**Comment by TBonnin** [comment]
Added a test https://github.com/guardian/frontend/pull/13015/files#diff-9dc855c91aacd716cf340f723deb3571R132 🎯

**Pull request merged**